tests/test_cityscapes_format.py

```diff
@@ -578,6 +578,218 @@ def categories(self):
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_dataset_with_specific_labelmap_with_bg_class(self):
+        class SrcExtractor(TestExtractorBase):
+            def __iter__(self):
+                yield DatasetItem(
+                    id=1,
+                    media=Image(data=np.ones((1, 5, 3))),
+                    annotations=[
+                        Mask(
+                            np.array([[1, 0, 0, 1, 1]]),
+                            label=1,
+                            id=1,
+                            attributes={"is_crowd": False},
+                        ),
+                        Mask(
+                            np.array([[0, 1, 1, 0, 0]]),
+                            label=2,
+                            id=2,
+                            attributes={"is_crowd": False},
+                        ),
+                    ],
+                )
+
+            def categories(self):
+                label_map = OrderedDict()
+                label_map["background"] = (0, 0, 0)
+                label_map["label_1"] = (1, 2, 3)
+                label_map["label_2"] = (3, 2, 1)
+                return Cityscapes.make_cityscapes_categories(label_map)
+
+        output_label_map = OrderedDict()
+        output_label_map["label_1"] = (4, 5, 6)
+        output_label_map["background"] = (1, 2, 3)
+        output_label_map["label_2"] = (2, 5, 7)
+
+        class DstExtractor(TestExtractorBase):
+            def __iter__(self):
+                yield DatasetItem(
+                    id=1,
+                    media=Image(data=np.ones((1, 5, 3))),
+                    annotations=[
+                        Mask(
+                            np.array([[1, 0, 0, 1, 1]]),
+                            attributes={"is_crowd": False},
+                            id=1,
+                            label=self._label("label_1"),
+                        ),
+                        Mask(
+                            np.array([[0, 1, 1, 0, 0]]),
+                            attributes={"is_crowd": False},
+                            id=2,
+                            label=self._label("label_2"),
+                        ),
+                    ],
+                )
+
+            def categories(self):
+                label_map = OrderedDict()
+                label_map["background"] = (1, 2, 3)  # must be moved to the idx 0
+                label_map["label_1"] = (4, 5, 6)
+                label_map["label_2"] = (2, 5, 7)
+                return Cityscapes.make_cityscapes_categories(label_map)
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(
+                SrcExtractor(),
+                partial(CityscapesConverter.convert, label_map=output_label_map, save_media=True),
+                test_dir,
+                target_dataset=DstExtractor(),
+            )
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_dataset_with_specific_labelmap_with_nonzero_bg_class(self):
```

Comment:
How is this test different from the previous one? It seems like it's the same thing except with `a` instead of `label_1`.

Reply:
The problem tested is that after label sorting the background class may be not at index 0, this test reproduces it.

Reply:
Okay, but couldn't that be integrated into the previous test? Just add a comment that the label names are chosen so that "background" is not alphabetically first, in order to test this case.

Reply:
That's much better, but I still don't see why these two tests couldn't be merged.

Reply:
Maybe they could, but they check different things. I'd split them further, instead - to check for sorting and for background insertion.

Reply:
Okay.